src/java/org/apache/cassandra/cql3/statements/UpdateStatement.java

```diff
@@ -93,7 +100,9 @@ public void addUpdateForKey(PartitionUpdate.Builder updateBuilder, Clustering<?>
             for (int i = 0, isize = updates.size(); i < isize; i++)
                 updates.get(i).execute(updateBuilder.partitionKey(), params);
 
-            updateBuilder.add(params.buildRow());
+            Row row = params.buildRow();
+            evaluateConstraintsForRow(row);
```

Comment:
@bbotella `metadata` can hold if there are some columns on constraints, when not, we can just skip call this method so we wont iterate over anything.